### ISSUE TITLE
toBeDisabled matcher proposal

### ISSUE THREAD

**Issue by callada** [gnapse/jest-dom#18]
### Describe the feature you'd like:

I would like to have a matcher that would check if the form element is disabled from user's perspective. 
In this case, simply checking the 'disabled' attribute is not enough. 

### Describe alternatives you've considered:

In many cases `.toHaveAttribute('disabled')` is sufficient alternative. 
But I think that the dedicated matcher would be less dependent on the implementation details

### Teachability, Documentation, Adoption, Migration Strategy:

Examples and comparison with `toHaveAttribute` matcher.

```javascript
// in simple cases toHaveAttribute and toBeDisabled behave the same

// ...
// <button disabled={true}>CLICK</button>
expect(getByText(container, 'CLICK')).toHaveAttribute('disabled')
expect(getByText(container, 'CLICK')).toBeDisabled()
// ...
```

```javascript
// toBeDisabled can detect when a disabled state is inherited

// ...
// <fieldset disabled={true}><button>CLICK</button></fieldset>
expect(getByText(container, 'CLICK')).not.toHaveAttribute('disabled')
expect(getByText(container, 'CLICK')).toBeDisabled()
// ...
```

```javascript
// 'a' element is not disabled, even if it has the 'disabled' attribute

// ...
// <a href="http://..." disabled={true}>CLICK</a>
expect(getByText(container, 'CLICK')).toHaveAttribute('disabled')
expect(getByText(container, 'CLICK')).not.toBeDisabled()
// ...
```

**Comment by kentcdodds** [comment]
I think this would be great 👍

**Comment by callada** [comment]
I'm happy that you like the idea. I can send PR for this feature.

**Comment by gnapse** [comment]
This is great. I did not think before about disabled being inherited by child elements, so it makes sense to make it easy and declarative to check for that.

I have one question though: why this?

```js
// 'a' element is not disabled, even if it has the 'disabled' attribute
```

I assume it is documented somewhere, so I'd like to learn about that. Why this special treatment with `a` elements?

**Comment by callada** [comment]
There are only 7 elements that can be [actually disabled](https://html.spec.whatwg.org/multipage/semantics-other.html#disabled-elements). Other elements are not disabled by the `disabled` attribute, `a` element is not special here.

**Comment by smacpherson64** [comment]
Implemented by @callada in #19
